src/robotide/publish/messages.py

```diff
@@ -64,16 +70,8 @@ def publish(self):
         implementation, if any of the listeners raises an exception, subsequent
         listeners will not get the notification.
         """
-        try:
-            self._publish(self)
-        except Exception as err:
-            self._publish(RideLogException(
-                message='Error in publishing message: ' + str(err),
-                exception=err, level='ERROR'))
-            raise # DEBUG on Python3 we have a loop
-
-    def _publish(self, msg):
-        publisher.PUBLISHER.publish(msg.topic, msg)
+        from robotide.publish.publisher import PUBLISHER
```

Comment:
Please don't do imports inside of the method.

Reply:
Sometimes internal importing is the only solution to solve circular imports.

Reply:
Yes, because there's a circular import if we put the import in the file head.

Reply:
ok